modules/git/git.go

```diff
@@ -55,31 +52,46 @@ func log(format string, args ...interface{}) {
 	}
 }
 
-// BinVersion returns current Git version from shell.
-func BinVersion() (string, error) {
-	if len(gitVersion) > 0 {
-		return gitVersion, nil
+// LocalVersion returns current Git version from shell.
+func LocalVersion() (*version.Version, error) {
+	if err := LoadGitVersion(); err != nil {
+		return nil, err
+	}
+	return gitVersion, nil
+}
+
+// LoadGitVersion returns current Git version from shell.
+func LoadGitVersion() error {
+	versionLock.RLock()
+	if gitVersion != nil {
+		versionLock.RUnlock()
+		return nil
 	}
+	versionLock.Lock()
```

Comment:
Need to RUnlock before Lock meaning you need to check gitVersion is not nil again

Reply:
was faster :)